lib/trust_is_risk.js

```diff
@@ -0,0 +1,202 @@
+//      
+var bcoin = require('bcoin');
+var assert = require('assert');
+var Address = bcoin.primitives.Address;
+
+                      // base58 bitcoin address
+                    
+                
+             
+                 
+                 
+                                                               
+ 
+                               
+                     
+
+class TrustIsRisk {
+                       
+  
+                 
+                        
+                            
+      
+   
+
+               
+                                 
+   
+
+  constructor(node                 ) {
+    this.node = node;
+    this.directTrust = {};
+    this.TXToTrust = {};
+
+    this.node.on('tx', this.addTX.bind(this));
+  }
+
+  getDirectTrust(from         , to         )          {
+    if (!this.directTrust.hasOwnProperty(from)) return 0;
+    if (!this.directTrust[from].hasOwnProperty(to)) return 0;
+    return this.directTrust[from][to];
+  }
+
+  // Attempts to parse a bitcoin transaction as a trust change and adds it to the trust network
+  // if successful.
+  // Returns true if the transaction is a TIR transaction and was successfully added to the
+  // network, false otherwise.
+  // Throws an error if the transaction was processed earlier.
+  addTX(tx           )           {
+    var txHash = tx.hash().toString('hex');
+    if (this.TXToTrust.hasOwnProperty(txHash)) {
+      throw new Error('Duplicate TX: Transaction with hash ' + txHash + ' has been seen again.');
+    }
+
+    var trustChanges = this.getTrustChanges(tx);
+    if (trustChanges.length === 0) return false;
+    else {
```

Comment:
I disagree with this: I use else after return only sometimes, when it enhances readability. This is usually when I want to distinguish between two cases, none of which are erroneous. I don't use else after return in initial error checking cases.

In this particular case, I'm using else exactly because an empty trust changes list is not error, but one of the two things that can happen: either this tx is a TIR transaction, or it's not.